robottelo/ui/role.py

```diff
@@ -83,22 +83,48 @@ def get_resources(self, role_name):
         :param role_name: String with role name.
         :return: List of strings with resource names.
         """
-        self.search_and_click(role_name)
-        self.click(tab_locators['roles.tab_filters'])
+        self.search(role_name)
+        self.click(locators['roles.filters_button'] % (role_name, role_name))
+        # make sure The role filter page is loaded
+        self.wait_until_element(locators['role_filters.title'])
         resources = [
             resource.text for resource in
-            self.find_elements(locators['roles.resources'])
+            self.find_elements(locators['role_filters.resources'])
         ]
-        next_ = self.find_element(locators["roles.filters.pagination_next"])
+        next_ = self.find_element(locators['role_filters.pagination_next'])
         while next_:
             self.click(next_)
-            next_ = self.find_element(
-                locators["roles.filters.pagination_next"])
+            self.wait_until_element(locators["role_filters.results_ready"])
             resources.extend(
                 [resource.text for resource in
-                    self.find_elements(locators['roles.resources'])])
+                 self.find_elements(locators['role_filters.resources'])])
+            next_ = self.find_element(locators['role_filters.pagination_next'])
         return resources
 
+    def _get_page_resources_permissions(self, resource_types, permissions):
+        """Retrieve the permissions for each resource type in resource_types
+        on Role Filters page.
+
+         :param resource_types: the list of resource types to check on the
+            current page.
+        :param  permissions: a dict store to register the resource permissions
+            found.
+
+        Note: The resource is removed from resource_types when found
+        """
+        self.wait_until_element(locators["role_filters.results_ready"])
+        resources = [
+            resource.text for resource in
+            self.find_elements(locators['role_filters.resources'])
+            ]
+        for res_type in resources:
+            if res_type in resource_types:
+                perm_element = self.find_element(
+                    locators['role_filters.permissions'] % res_type)
+                if perm_element:
+                    permissions[res_type] = perm_element.text.split(', ')
```

Comment:
Could you please describe what's going on here? Previously the table contained 1 resource type per line, and for each resource type all the permissions were shown in one cell, separated by comma. That's why we had lines like this (`permissions[res_type] = perm_element.text.split(', ')`).
Now, as far as i can see, we have separate line per each permission, making this code invalid. The fix should have been something like grabbing all the cells where preceding cell (resource type) is equal to `res_type`. 
For me locally the fix is only 3 lines long, but here i can see one `get_permissions` rewritten and extra private method created. What's the reason behind this?